### ISSUE TITLE
App fails to build with Turbopack loader

### ISSUE THREAD

**Issue by racsogrok** [vercel/next.js#48140]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: x64
  Version: Darwin Kernel Version 22.3.0: Mon Jan 30 20:42:11 PST 2023; root:xnu-8792.81.3~2/RELEASE_X86_64
Binaries:
  Node: 16.19.1
  npm: 8.19.3
  Yarn: 1.22.19
  pnpm: N/A
Relevant packages:
  next: 13.3.1-canary.3
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
```

### Which example does this report relate to?

with-turbopack-loaders

### What browser are you using? (if relevant)

Safari 18614.4.6.1.6

### Describe the Bug

The Next.js example with a Turbopack loader does not build successfully with `npm run build`.

### Expected Behavior

`npm run build` successfully builds the applications even with a custom Turbopack loader.

### To Reproduce

I created a Next.js application with the following script:

```
npx create-next-app@latest --example with-turbopack-loaders
```

That used the `with-turbopack-loaders` example to scaffold the app.

Running `npm run dev` works successfully, and I'm able to interact with the application in my browser.

When I tried to build the app (via `npm run build`), the build fails with the following error:

```
[    ] info  - Generating static pages (0/3)Error: Minified React error #130; visit https://reactjs.org/docs/error-decoder.html?invariant=130&args[]=object&args[]= for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
```

When I opened the error link provided, it specified the full error as:

```
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.
```

In summary, I'm not able to build the app with a custom Turbopack loader.

**Comment by rtritto** [comment]
I got same error on some pages if I add `"type": "module"`.

**Comment by tntchn** [comment]
I found the error was probably caused by a feature in turebopack [vercel:next.js#48947](https://github.com/vercel/next.js/pull/48947)

From next 13.3.2, turbopack would try to resolve the dimension of a image maybe for

`<Image src='/public/image.svg' />`

however when @svgr/webpack was used in the turbopack custom loader, the code
```typescript
import SomeImage from '../image.svg'
function Component() {
  return <SomeImage />
}
```
will pack `SomeImage` to a function `SvgImage()` therefore if I only try to get `const img = SomeImage`
```
error - [image] [project]/vercel.svg  Processing image failed
  Failed to parse svg source code for image dimensions
  
  Caused by:
  - Source code does not contain a <svg> root element
 ```
[here?](https://github.com/vercel/turbo/blob/a00ba87b486db48038e1d6e74775a29b920ef03b/crates/turbopack-image/src/process/svg.rs#L81C8-L84)

The error logs before 13.3.4 provides more useful information
```
Failed to parse svg source code for image dimensions

Caused by:
- Source code does not contain a <svg> root element

Debug info:
- Execution of get_from_source failed
- Execution of resolve_source_request failed
- Execution of NodeRenderGetContentResult::get failed
- Execution of render_static failed
- Execution of get_renderer_pool failed
- Execution of emit failed
- Execution of internal_assets failed
- Execution of separate_assets failed
- Execution of primary_referenced_assets failed
- Execution of NodeJsBootstrapAsset::references failed
- Execution of DevChunkingContext::evaluated_chunk_group failed
- Execution of EcmascriptChunk::parallel_chunks failed
- Execution of ecmascript_chunk_content_single_entry failed
- Execution of ChunkGroupFilesAsset::references failed
- Execution of DevChunkingContext::evaluated_chunk_group failed
- Execution of EcmascriptChunk::parallel_chunks failed
- Execution of ecmascript_chunk_content_single_entry failed
- Execution of EcmascriptModuleAsset::references failed
- Execution of turbopack_ecmascript::EcmascriptModuleAssetVc::failsafe_analyze failed
- Execution of analyze_ecmascript_module failed
- Execution of parse failed
- Execution of StructuredImageSourceAsset::content failed
- Execution of get_meta_data failed
- Failed to parse svg source code for image dimensions
- Source code does not contain a <svg> root element
```

**Comment by tntchn** [comment]
I don't know if there is a good way to ask turbopack not getting the dimension of my svgs but if I rename the file extension to `.someother` and change my next config to 
```js
experimental: {
  turbo: {
    rules: {
      '*.someother': ['@svgr/webpack'],
    }
  }
}
```
my app can run as normal for me to test other things

**Comment by Vsion** [comment]
> I don't know if there is a good way to ask turbopack not getting the dimension of my svgs but if I rename the file extension to `.someother` and change my next config to
> 
> ```js
> experimental: {
>   turbo: {
>     rules: {
>       '*.someother': ['@svgr/webpack'],
>     }
>   }
> }
> ```
> 
> my app can run as normal for me to test other things

it works in v14, really
out of my mind.

**Comment by ForsakenHarmony** [comment]
see https://github.com/vercel/turbo/issues/4832
